src/orchestrator.js

```diff
@@ -80,44 +88,130 @@ function execPreCommands(config) {
   });
 }
 
-function genearateSpecsForMachines(config) {
-  let specs =
-    config.specs.length > 0 ? config.specs : sh.ls(config.specsHomePath);
-  let [start, end] = [0, 0];
+function getListOfSpecs(config, browser) {
+  let existingSpecs = [];
+
+  if (config.specs.length > 0) {
+    existingSpecs = config.specs
+  } else {
+    existingSpecs = sh.ls(config.specsHomePath);
+  }
+
+  if (checkFileIsExisting(config.specsExecutionTimePath)) {
+    let specsExecutionTime = parseJsonFile(config.specsExecutionTimePath);
+    let browserSpecs = orderBasedOnBrowserDuration(
+      specsExecutionTime,
+      browser
+    ).map((item) => item.specName);
+
+    let specs = browserSpecs.filter((spec) => existingSpecs.includes(spec));
+    specs = [
+      ...specs,
+      ...existingSpecs.filter((item) => !specs.includes(item)),
+    ];
+
+    return specs;
+  } else {
+    return existingSpecs;
+  }
+}
+
+function removeEmpty(arrays) {
+  let results = [];
+  arrays.forEach((array) => {
+    if (array.length > 0) results.push(array.filter((item) => item !== ""));
+  });
+  return results;
+}
+
+function splitSpecsOverMachines(specs, config) {
+  let noOfMachines = config.parallelizm * config.browsers.length;
   let specsForMachines = [];
 
-  if (config.parallelizm > specs.length) {
-    config.parallelizm = specs.length;
+  for (let i = 0; i < noOfMachines; i++) {
+    specsForMachines.push([]); // [ [], [], [] ..]
+  }
+
+  let _cycles = 0;
+  while (specs.length > 0) {
+    for (let i = 0; i < noOfMachines; i++) {
+      if (specs.length == 0) break;
+      _cycles % 2
```

Comment:
What if I'd like to split specs into 4 machines?

Reply:
it will work too.
for the 1st and 3rd machines, it will start chrome then firefox
for the 2nd and 4th machines, it will start firefox then chrome